BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -125,6 +136,24 @@ public string Id
             get; private set;
         }
 
+        /// <summary>
+        /// Gets a value indicating whether flush is async.
+        /// </summary>
+        public bool ManualFlushAsyncFlag
```

Comment:
Boolean properties should be named `Is...` or `Has...`
e.g. `IsFlushAsync` or `HasFlushTask`

I also think this can be implemented completely over the `TaskCompletionSource` without using a separate (backing) field.

i.e.
```csharp
public bool HasFlushTask =>
    FlushTaskCompletionSource != null && !FlushTaskCompletionSource.Task.IsCompleted;
```
Then, in the `SetFlushTaskCompletionSourceResult()` method, you don't set this property to false, you just SetResult/SetCancel/SetException as necessary.